parser/internal/parserv2/parser.go

```diff
@@ -801,7 +879,20 @@ func (p *parser) parse() (ast.Expression, error) {
 	}
 	p.unscan()
 
-	var err error
+	// Parse top level fragments
+	fragments, err := p.parseFragments()
+	if err != nil {
+		return nil, err
+	}
+
+	// Keep all fragments as a map in the parser to be referenced later when parsing the main query
+	p.fragments = make(map[string]*ast.Fragment)
+	for _, fragment := range fragments {
```

Comment:
We can drop this loop if `parseFragments` appends to the map when it have parsed a fragment?